packages/uploadthing/src/client.ts

```diff
@@ -102,25 +101,25 @@ export const genUploader = <TRouter extends FileRouter>(
       UploadFilesOptions<TRouter, TEndpoint, TSkipPolling>,
       keyof GenerateUploaderOptions
     >,
-  ) => {
-    const layer = Layer.succeed(fetchContext, {
-      fetch: globalThis.fetch.bind(globalThis),
-      baseHeaders: {
-        "x-uploadthing-version": UPLOADTHING_VERSION,
-        "x-uploadthing-api-key": undefined,
-        "x-uploadthing-fe-package": initOpts.package,
-        "x-uploadthing-be-adapter": undefined,
-      },
-    });
-
-    return uploadFilesInternal<TRouter, TEndpoint, TSkipPolling>(endpoint, {
+  ) =>
+    uploadFilesInternal<TRouter, TEndpoint, TSkipPolling>(endpoint, {
       ...opts,
       url: resolveMaybeUrlArg(initOpts?.url),
       package: initOpts.package,
       // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
       input: (opts as any).input as inferEndpointInput<TRouter[TEndpoint]>,
-    }).pipe(Effect.provide(layer), Effect.runPromise);
-  };
+    }).pipe(
+      Effect.provideService(FetchContext, {
```

Comment:
You can use Effect.provideService instead of a Layer here. Layer's are better for composing lots of services together, rather than providing simply one-off deps.